src/components/[guild]/activity/ActivityLogAction/components/RewardTag.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  ChakraProps,
+  forwardRef,
+  Tag,
+  TagLabel,
+  TagLeftIcon,
+  TagProps,
+  Tooltip,
+} from "@chakra-ui/react"
+import { IconProps } from "phosphor-react"
+import platforms from "platforms/platforms"
+import { PlatformType } from "types"
+import { useActivityLog } from "../../ActivityLogContext"
+import { useActivityLogFilters } from "../../ActivityLogFiltersBar/components/ActivityLogFiltersContext"
+
+type Props = {
+  name?: string
+  icon?: (props: IconProps) => JSX.Element
+  colorScheme?: ChakraProps["color"]
+} & Omit<TagProps, "colorScheme">
+
+const RewardTag = forwardRef<Props, "span">(
```

Comment:
Why is the abstraction so general, instead of just needing the type and id, and handling the rest? Providing data for the current props from platforms.ts (and a `platformId === PlatformType.DISCORD` condition) seems to be repeated 3 times now instead of just being here, but I might be missing something

Reply:
Oh, nice catch, refactored all these tags in e61a83ecf54aa148b5afb59f9775b4766e259754